### PR TITLE
prevent reloading of configurations containing non reloadable plugins

### PR DESCRIPTION
**Pull request by jsvd**
fixes https://github.com/elastic/logstash/issues/4862

### CODE REVIEW

**Comment by suyograo** [comment]
```
suyog@machine:~/ws/elastic/logstash (pr/4884)$ bin/logstash -f ~/ws/elastic/conf/complete.conf --auto-reload
Connection refused {:class=>"Manticore::SocketException", :level=>:error}
Connection refused {:class=>"Manticore::SocketException", :level=>:error}
Connection refused {:class=>"Manticore::SocketException", :level=>:error}
Connection refused {:class=>"Manticore::SocketException", :level=>:error}
Connection refused {:class=>"Manticore::SocketException", :level=>:error}
Connection refused {:class=>"Manticore::SocketException", :level=>:error}
Connection refused {:class=>"Manticore::SocketException", :level=>:error}
Connection refused {:class=>"Manticore::SocketException", :level=>:error}
Logstash was not able to load configuration since it does not support
dynamic reloading and -r or --auto-reload flag was enabled {:pipeline_id=>"main", :configuration=>"input {\n  stdin { }\n}\n\nfilter {\n  grok {\n    match => {\n      \"message\" => '%{IPORHOST:clientip} %{USER:ident} %{USER:auth} \\[%{HTTPDATE:timestamp}\\] \"%{WORD:verb} %{DATA:request} HTTP/%{NUMBER:httpversion}\" %{NUMBER:response:int} (?:-|%{NUMBER:bytes:int}) %{QS:referrer} %{QS:agent}'\n    }\n  }\n\n  date {\n    match => [ \"timestamp\", \"dd/MMM/YYYY:HH:mm:ss Z\" ]\n    locale => en\n  }\n\n  geoip {\n    source => \"clientip\"\n  }\n\n  useragent {\n    source => \"agent\"\n    target => \"useragent\"\n  }\n}\n\noutput {\n  stdout { codec => rubydebug }\n  elasticsearch {\n    hosts => \"localhost\"\n  }\n}\n\n", :plugins=>[LogStash::Inputs::Stdin], :level=>:error}
```

I don't think we should dump the entire config in the stdout. Users have pretty big configs and the error message just hides this.

Also, I tested on 5.0.0 and i get these additional manticore issue - related to webserver? /cc @purbon

File: logstash-core/lib/logstash/agent.rb [hunk 1]
```diff
@@ -77,6 +77,13 @@ def execute
   def register_pipeline(pipeline_id, settings)
     pipeline = create_pipeline(settings.merge(:pipeline_id => pipeline_id, :metric => metric))
     return unless pipeline.is_a?(LogStash::Pipeline)
+    if @auto_reload && pipeline.non_reloadable_plugins.any?
+      @logger.error(I18n.t("logstash.agent.non_reloadable_config_register"),
+                    :pipeline_id => pipeline_id,
+                    :configuration => pipeline.config_str,
```
Comment (suyograo) [inline logstash-core/lib/logstash/agent.rb]: There's no need to dump the entire config here. It just hides the original error message

File: logstash-core/lib/logstash/agent.rb [hunk 2]
```diff
@@ -77,6 +77,13 @@ def execute
...
+                    :plugins => pipeline.non_reloadable_plugins.map(&:class))
```
Comment (suyograo) [inline logstash-core/lib/logstash/agent.rb]: Just this is good

File: logstash-core/lib/logstash/agent.rb [hunk 3]
```diff
@@ -109,6 +116,12 @@ def node_uuid
...
   end
 
+  def running_pipelines?
+    @upgrade_mutex.synchronize do
+      @pipelines.select {|pipeline_id, _| running_pipeline?(pipeline_id) }.any?
+    end
+  end
+
   private
   def start_webserver
```
Comment (suyograo) [inline logstash-core/lib/logstash/agent.rb]: Check the error we are getting from manticore, maybe related to starting webserver

**Comment by suyograo** [comment]
Tested both scenarios, works well.

FWIW, Manticore issues are not related to this PR.

**Comment by jsvd** [comment]
I need to do another PR against 2.x/2.3

**Pull request closed without merging**